src/frequenz/channels/_latest_value_cache.py

```diff
@@ -78,34 +134,53 @@ def unique_id(self) -> str:
         """The unique identifier of this instance."""
         return self._unique_id
 
-    def get(self) -> T_co:
+    def get(self, key: HashableT | None = None) -> T_co:
         """Return the latest value that has been received.
 
         This raises a `ValueError` if no value has been received yet. Use `has_value` to
         check whether a value has been received yet, before trying to access the value,
         to avoid the exception.
 
+        Args:
+            key: An optional key to retrieve the latest value for that key. If not
+                provided, it retrieves the latest value received overall.
+
         Returns:
             The latest value that has been received.
 
         Raises:
             ValueError: If no value has been received yet.
         """
+        if key is not None:
+            if key not in self._latest_value_by_key:
+                raise ValueError(f"No value received for key: {key!r}")
+            return self._latest_value_by_key[key]
```

Comment:
If you use a sentinel, you can probably also get rid of the "global" `self._latest_value` and use `self._latest_value_by_key[_Sentinel]` instead as the global value, it could even simplify the code quite a bit as you get rid of the special casing.

Reply:
I just tried this and neither the number of if statements nor the number of lines come down.  In addition to that we're having to hash an extra param.  So I don't see the point.

It would have been different, if the place holder in the init wasn't for a key /function/, but rather for a key, I guess, not sure.